src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -297,6 +326,20 @@ private Collection<Query> toQueries(Collection<QueryBuilder> queryBuilders, Quer
         return queries;
     }
 
+    private static void validatePaginationDepth(final int paginationDepth, final QueryShardContext queryShardContext) {
+        if (paginationDepth < LOWER_BOUND_OF_PAGINATION_DEPTH) {
+            throw new IllegalArgumentException(String.format(Locale.ROOT, "pagination_depth should be greater than 0"));
```

Comment:
we should either hardcode '0' for the if condition check, or use  `LOWER_BOUND_OF_PAGINATION_DEPTH` constant in the exception message, this way it will be consistent